xlml/apis/task.py

```diff
@@ -410,6 +433,24 @@ def run(self) -> DAGNode:
       provision >> run_model >> post_process >> clean_up
     return group
 
+  def run_with_gcs_name_generation(self) -> DAGNode:
+    with TaskGroup(
+        group_id=self.task_test_config.benchmark_id, prefix_group_id=True
+    ) as group:
+      gcs_location = name_format.generate_gcs_file_location(
```

Comment:
nit: can `gcs_location` move into `provision`?

Reply:
Yes, the trade off is that we will introduce an extra return in `provision` function but can get rid of `gcs_location` stage. Are we fine with that?

Reply:
I think moving the `gcs_location` to `provision` stage makes the code clumpy:
```
  def provision(self) -> Tuple[DAGNode, airflow.XComArg, airflow.XComArg, airflow.XComAry]:
    with TaskGroup(group_id="provision") as group:
      with TaskGroup(group_id="initialize"):
        tpu_name = tpu.generate_tpu_name(
            self.task_test_config.benchmark_id, self.tpu_name_env_var
        )
        ssh_keys = ssh.generate_ssh_keys()

      queued_resource_op, queued_resource_name = tpu.create_queued_resource(
          tpu_name,
          self.task_gcp_config,
          ssh_keys,
          self.tpu_create_timeout,
          self.task_test_config,
      )

      gcs_location = name_format.generate_gcs_file_location(
          self.task_test_config.benchmark_id
      )

      queued_resource_op >> tpu.ssh_tpu.override(task_id="setup")(
          queued_resource_name,
          # TODO(wcromar): remove split
          self.task_test_config.setup_script,
          ssh_keys,
          self.all_workers,
      )

    return group, queued_resource_name, ssh_keys, gcs_location 
```
Not everyone wants the gcs_location here. We can do refactor later together with other customized `run_with...` functions.

Reply:
Is there any harm in always generating a GCS location name? I actually think this would be helpful. The old system used to pass in a unique GCS `${MODEL_DIR}` in every case, and most (but not all) tests used it. This would let us cut out yet another variant of `run`

Reply:
Oh, if we want to let user specify the gcs location as a standard routine, then adding this to provision should be good. Let me make the update.